src/probnum/filtsmooth/statespace/transition.py

```diff
@@ -1,221 +1,355 @@
+        Dict
+            Information about the forward pass. Can for instance contain a `gain` key, if `compute_gain` was set to `True` (and if the transition supports this functionality).
         """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def transition_rv(
-        self,
-        rv: "RandomVariable",
-        start: float,
-        stop: Optional[float] = None,
-        step: Optional[float] = None,
-        _diffusion: Optional[float] = 1.0,
-        _linearise_at: Optional[RandomVariable] = None,
-    ) -> (RandomVariable, Dict):
-        """Transition a random variable from time :math:`t` to time
-        :math:`t+\\Delta t`.
+    def forward_realization(
+        self, real, t, dt=None, compute_gain=False, _diffusion=1.0, _linearise_at=None
+    ):
+        r"""Forward-pass of a realization of a state, according to the transition. In other words, return a description of
 
-        For random variable :math:`x_t`, it returns the random variable defined by
+        .. math:: p(\mathcal{G}_t[x(t)] \,|\, x(t)=\xi),
 
-        .. math:: x_{t + \\Delta t} \\sim p(x_{t + \\Delta t}  | x_t) .
+        for some realization :math:`\xi`.
 
-        This returns a random variable where the parametrization depends on the paramtrization of :math:`x_t`.
-        This is different to :meth:`transition_rv` which computes the parametrization
-        of :math:`x_{t + \\Delta t}` based on a realization of :math:`x_t`.
+        Parameters
+        ----------
+        real
+            Realization :math:`\xi` of the random variable :math:`x(t)` that describes the current state.
+        t
+            Current time point.
+        dt
+            Increment :math:`\Delta t`. Ignored for discrete-time transitions.
+        compute_gain
+            Flag that indicates whether the expected gain of the forward transition shall be computed. This is important if the forward-pass
+            is computed as part of a forward-backward pass, as it is for instance the case in a Kalman update.
+        _diffusion
+            Special diffusion of the driving stochastic process, which is used internally.
+        _linearise_at
+            Specific point of linearisation for approximate forward passes (think: extended Kalman filtering). Used internally for iterated filtering and smoothing.
 
-        Nb: Think of transition as a verb, i.e. this method "transitions" a random variable.
+        Returns
+        -------
+        RandomVariable
+            New state, after applying the forward-pass.
+        Dict
+            Information about the forward pass. Can for instance contain a `gain` key, if `compute_gain` was set to `True` (and if the transition supports this functionality).
+        """
+        raise NotImplementedError
 
+    @abc.abstractmethod
+    def backward_rv(
+        self,
+        rv_obtained,
+        rv,
+        rv_forwarded=None,
+        gain=None,
+        t=None,
```

Comment:
Only when forward_rv is called internally to compute gain and forwarded rv. If not, this information is contained in the gain and the forwarded rv. I suppose it is a trade-off between (`rv_forwarded, gain`) and (`t, dt`) when it comes to positioning. Would you change the orders?

Reply:
I don't really have an opinion on this, I just wanted to question it in case this was not that intentional. Since you have a better idea on how the code is used, so feel free to keep the current order :)